events/codepipeline_cloudwatch.go

```diff
@@ -0,0 +1,106 @@
+package events
+
+import (
+	"time"
+)
+
+const (
+	CodePipelineEventSource              = "aws.codepipeline"
+	CodePipelineExecutionEventDetailType = "CodePipeline Pipeline Execution State Change"
+	CodePipelineActionEventDetailType    = "CodePipeline Action Execution State Change"
+	CodePipelineStageEventDetailType     = "CodePipeline Stage Execution State Change"
+)
+
+type CodePipelineStageState string
+
+const (
+	CodePipelineStageStateStarted   CodePipelineStageState = "STARTED"
+	CodePipelineStageStateSucceeded CodePipelineStageState = "SUCCEEDED"
+	CodePipelineStageStateResumed   CodePipelineStageState = "RESUMED"
+	CodePipelineStageStateFailed    CodePipelineStageState = "FAILED"
+	CodePipelineStageStateCanceled  CodePipelineStageState = "CANCELED"
+)
+
+type CodePipelineState string
+
+const (
+	CodePipelineStateStarted    CodePipelineState = "STARTED"
+	CodePipelineStateSucceeded  CodePipelineState = "SUCCEEDED"
+	CodePipelineStateResumed    CodePipelineState = "RESUMED"
+	CodePipelineStateFailed     CodePipelineState = "FAILED"
+	CodePipelineStateCanceled   CodePipelineState = "CANCELED"
+	CodePipelineStateSuperseded CodePipelineState = "SUPERSEDED"
+)
+
+type CodePipelineActionState string
+
+const (
+	CodePipelineActionStateStarted   CodePipelineActionState = "STARTED"
+	CodePipelineActionStateSucceeded CodePipelineActionState = "SUCCEEDED"
+	CodePipelineActionStateFailed    CodePipelineActionState = "FAILED"
+	CodePipelineActionStateCanceled  CodePipelineActionState = "CANCELED"
+)
+
+// CodePipelineEvent is documented at:
+// https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/EventTypes.html#codepipeline_event_type
+type CodePipelineCloudWatchEvent struct {
+	// Version is the version of the event's schema.
+	Version string `json:"version"`
+
+	// ID is the GUID of this event.
+	ID string `json:"id"`
+
+	// DetailType informs the schema of the Detail field. For deployment state-change
+	// events, the value should be equal to CodePipelineDeploymentEventDetailType.
+	// For instance state-change events, the value should be equal to
+	// CodePipelineInstanceEventDetailType.
+	DetailType string `json:"detail-type"`
+
+	// Source should be equal to CodePipelineEventSource.
+	Source string `json:"source"`
+
+	// AccountID is the id of the AWS account from which the event originated.
+	AccountID string `json:"account"`
+
+	// Time is the event's timestamp.
+	Time time.Time `json:"time"`
+
+	// Region is the AWS region from which the event originated.
+	Region string `json:"region"`
+
+	// Resources is a list of ARNs of CodePipeline applications and deployment
+	// groups that this event pertains to.
+	Resources []string `json:"resources"`
+
+	// Detail contains information specific to a deployment event.
+	Detail CodePipelineEventDetail `json:"detail"`
+}
+
+type CodePipelineEventDetail struct {
+	Pipeline string `json:"pipeline"`
+
+	// From live testing this is always int64 not string as documented
```

Comment:
suspicious... example in docs appear inconsistent on this one across "Pipeline Execution State Change"/"Stage Execution State Change"/"Action Execution State Change"

Reply:
Ill just confirm this from real data as well as the one version left as a `string` on this [line](https://github.com/aws/aws-lambda-go/pull/247/files#diff-04cdfdd664e8f28c763446462d2dbfc266ef7a011fad402566b043d6ab69fdb3R48) before I request re-review